ethereum/ethstats/src/test/java/org/hyperledger/besu/ethstats/EthStatsServiceTest.java

```diff
@@ -343,4 +344,108 @@ public boolean failed() {
       }
     };
   }
+
+  @Test
+  public void shouldUseCustomReportInterval() {
+    // Test with custom 10-second interval
+    final EthStatsConnectOptions customIntervalOptions =
+        ImmutableEthStatsConnectOptions.builder()
+            .nodeName("besu-node")
+            .secret("secret")
+            .host("127.0.0.1")
+            .port(1111)
+            .contact("contact@test.net")
+            .ethStatsReportInterval(10)  // Custom 10-second interval
+            .build();
+
+    ethStatsService =
+        new EthStatsService(
+            customIntervalOptions,
+            blockchainQueries,
+            ethProtocolManager,
+            transactionPool,
+            miningCoordinator,
+            syncState,
+            vertx,
+            "clientVersion",
+            genesisConfigOptions,
+            p2PNetwork);
+    when(p2PNetwork.getLocalEnode()).thenReturn(Optional.of(node));
+
+    final ArgumentCaptor<Handler<AsyncResult<WebSocket>>> webSocketCaptor =
+        ArgumentCaptor.forClass(Handler.class);
+
+    ethStatsService.start();
+
+    verify(webSocketClient, times(1))
+        .connect(any(WebSocketConnectOptions.class), webSocketCaptor.capture());
+    webSocketCaptor.getValue().handle(succeededWebSocketEvent(Optional.of(webSocket)));
+
+    final ArgumentCaptor<Handler<String>> textMessageHandlerCaptor =
+        ArgumentCaptor.forClass(Handler.class);
+    verify(webSocket, times(1)).textMessageHandler(textMessageHandlerCaptor.capture());
+
+    textMessageHandlerCaptor.getValue().handle("{\"emit\":[\"ready\"]}");
+
+    // Verify that scheduleFutureTaskWithFixedDelay is called with the custom 10-second interval
+    final ArgumentCaptor<Duration> initialDelayCaptor = ArgumentCaptor.forClass(Duration.class);
+    final ArgumentCaptor<Duration> intervalCaptor = ArgumentCaptor.forClass(Duration.class);
+    verify(ethScheduler, times(1))
+        .scheduleFutureTaskWithFixedDelay(
+            any(Runnable.class), 
+            initialDelayCaptor.capture(), 
+            intervalCaptor.capture());
+
+    // Verify the interval is 10 seconds (custom value)
+    assertThat(intervalCaptor.getValue()).isEqualTo(Duration.ofSeconds(10));
+    // Verify initial delay is 0 seconds
+    assertThat(initialDelayCaptor.getValue()).isEqualTo(Duration.ofSeconds(0));
+  }
+
+  @Test
+  public void shouldUseDefaultReportIntervalWhenNotSpecified() {
```

Comment:
Hello, I am **not maintainer** of besu code, just opened your PR as part of my learning process and noticed some stuff, **please feel free to ignore my comments** if you find them irrelevant 

1) could you please check is it possible to modify existing test like `shouldSendHelloMessage` that it run with default value instead of creating a new one?
It will allow to decrease a cost of maintenance, but I am not sure about these, just highlighting it

Reply:
@AliZDev-v0 prefer to keep tests single-purpose ie shouldSendHelloMessage should not test anything other than sending a hello message. If this test fails `shouldUseDefaultReportIntervalWhenNotSpecified` we know exactly what the problem is